src/coreclr/vm/clsload.cpp

```diff
@@ -2055,11 +1998,10 @@ VOID ClassLoader::Init(AllocMemTracker *pamTracker)
                              CrstAvailableClass,
                              CRST_REENTRANCY);
 
-    // This lock is taken within the classloader whenever we have to insert a new param. type into the table
-    // This lock also needs to be taken for a read operation in a GC_NOTRIGGER scope, thus the ANYMODE flag.
+    // This lock is taken within the classloader whenever we have to insert a new param. type into the table.
     m_AvailableTypesLock.Init(
-                                  CrstAvailableParamTypes,
-                                  (CrstFlags)(CRST_UNSAFE_ANYMODE | CRST_DEBUGGER_THREAD));
+                              CrstAvailableParamTypes,
+                              CRST_DEBUGGER_THREAD);
```

Comment:
@hoyosjs - If debugger only does lookups, then it will no longer take this look, then  `CRST_DEBUGGER_THREAD` is not needed. 
Can debugger load/publish types?

Reply:
I know we can't load assemblies, but it's possible that a FuncEval can load a type I believe.

Reply:
Is funceval running on debugger thread?   

I assumed that `CRST_DEBUGGER_THREAD` does not cover funceaval, since funceval could JIT and JIT does all kind of stuff (including loading assemblies).  is attributed to the debugger thread, then debugger thread is not different from anything else.
I am not very sure how that all works though.

 `CRST_DEBUGGER_THREAD` is not a big nuisance here. I was just not sure it is still necessary.

Reply:
I've tried removing `CRST_DEBUGGER_THREAD` and running both regular and diagnostics tests (with Chk bits) - everything runs as before. Not sure if that is enough proof that `CRST_DEBUGGER_THREAD` is unnecessary.

I just noticed that CRST_DEBUGGER_THREAD` is always paired with `CRST_UNSAFE_ANYMODE` (or sometimes `CRST_GC_NOTRIGGER_WHEN_TAKEN`), so since we are removing the other one, maybe we do not need `CRST_DEBUGGER_THREAD` either. 

But do not think `CRST_DEBUGGER_THREAD` is a big nuisance either way - it basically increments/decrements a counter that is checked in couple asserts.